extensions/binding/org.eclipse.smarthome.binding.ntp.test/src/test/java/org/eclipse/smarthome/binding/ntp/test/NtpOSGiTest.java

```diff
@@ -499,25 +499,18 @@ private void assertEventIsReceived(UpdateEventType updateEventType, String chann
         Configuration configuration = new Configuration();
         initialize(configuration, channelID, acceptedItemType, null, null);
 
-        eventSubscriberMock = new EventSubscriberMock();
-        registerService(eventSubscriberMock, EventSubscriber.class.getName());
+        EventSubscriber eventSubscriberMock = mock(EventSubscriber.class);
+        when(eventSubscriberMock.getSubscribedEventTypes()).thenReturn(Collections.singleton(ItemStateEvent.TYPE));
+        registerService(eventSubscriberMock);
 
         if (updateEventType.equals(UpdateEventType.HANDLE_COMMAND)) {
             ntpHandler.handleCommand(new ChannelUID("ntp:test:chan:1"), new StringType("test"));
         } else if (updateEventType.equals(UpdateEventType.CHANNEL_LINKED)) {
             ntpHandler.channelLinked(new ChannelUID("ntp:test:chan:1"));
         }
         waitForAssert(() -> {
-            assertTrue(eventSubscriberMock.isEventReceived);
+            verify(eventSubscriberMock).receive(ArgumentMatchers.any(Event.class));
```

Comment:
I am not such familiar with Mockito (shame on me)... Does this also work if `receive` is also called in fron of the `waitForAssert` block? Or rephrase: Does this verify check if the method has been called "before"?

Reply:
To my understanding, mockito simply records _all_ interactions and with the `verify(..)` call you simply make sure that _some_ matching interaction happened until now. So it doesn't block at this point and it also doesn't "eat up" one interaction if the `verify(...)` would be called twice.

By the way, if it would be relevant how many times a certain interaction happened, then there is `verify(mock, times(n))` etc.

Reply:
> By the way, if it would be relevant how many times a certain interaction happened, then there is `verify(mock, times(n))` etc. 

Ah, thanks.

Reply:
Argh 😒 

Minor (but important) correction:

```
Wanted 1 time:
-> at org.eclipse.smarthome.binding.ntp.test.NtpOSGiTest.lambda$5(NtpOSGiTest.java:512)
But was 4 times. Undesired invocation:
-> at sun.reflect.GeneratedMethodAccessor11.invoke(Unknown Source)
	at org.eclipse.smarthome.binding.ntp.test.NtpOSGiTest.lambda$5(NtpOSGiTest.java:512)
```

I just learned that `verify(mock)` is equivalent to `verify(mock, times(1))`. My assumption that it is the same as `verify(mock, atLeastOnce())` apparently was wrong. Your question was pretty much to the point. Sorry!